agent_build/common.py

```diff
@@ -0,0 +1,341 @@
+    for file_name in file_names:
+        matchers.append(re.compile(str(file_name)))
+
+    # Walk down the current directory.
+    for root, dirs, files in os.walk(dir_path.absolute()):
+        # See if any of the files at this level match any of the matchers.
+        for file_path in files:
+            remove_it = False
+            for matcher in matchers:
+                if matcher.match(file_path):
+                    remove_it = True
+            # Delete it if it did match.
+            if remove_it:
+                os.unlink(os.path.join(root, file_path))
+
+
+class BadChangeLogFormat(Exception):
+    pass
+
+def parse_date(date_str):
+    """Parses a date time string of the format MMM DD, YYYY HH:MM +ZZZZ and returns seconds past epoch.
+
+    Example of the format is: Oct 10, 2014 17:00 -0700
+
+    @param date_str: A string containing the date and time in the format described above.
+
+    @return: The number of seconds past epoch.
+
+    @raise ValueError: if there is a parsing problem.
+    """
+    # For some reason, it was hard to parse this particular format with the existing Python libraries,
+    # especially when the timezone was not the same as the local time zone.  So, we have to do this the
+    # sort of hard way.
+    #
+    # It is a little annoying that strptime only takes Sep for September and not Sep which is more common
+    # in US-eng, so we cheat here and just swap it out.
+    adjusted = date_str.replace("Sept", "Sep")
+
+    # Find the timezone string at the end of the string.
+    if re.search(r"[\-+]\d\d\d\d$", adjusted) is None:
+        raise ValueError(
+            "Value '%s' does not meet required time format of 'MMM DD, YYYY HH:MM +ZZZZ' (or "
+            "as an example, ' 'Oct 10, 2014 17:00 -0700'" % date_str
+        )
+
+    # Use the existing Python string parsing calls to just parse the time and date.  We will handle the timezone
+    # separately.
+    try:
+        base_time = time.mktime(time.strptime(adjusted[0:-6], "%b %d, %Y %H:%M"))
+    except ValueError:
+        raise ValueError(
+            "Value '%s' does not meet required time format of 'MMM DD, YYYY HH:MM +ZZZZ' (or "
+            "as an example, ' 'Oct 10, 2014 17:00 -0700'" % date_str
+        )
+
+    # Since mktime assumes the time is in localtime, we might have a different time zone
+    # in tz_str, we must manually added in the difference.
+    # First, convert -0700 to seconds.. the second two digits are the number of hours
+    # and the last two are the minute of minutes.
+    tz_str = adjusted[-5:]
+    tz_offset_secs = int(tz_str[1:3]) * 3600 + int(tz_str[3:5]) * 60
+
+    if tz_str.startswith("-"):
+        tz_offset_secs *= -1
+
+    # Determine the offset for the local timezone.
+    if time.daylight:
+        local_offset_secs = -1 * time.altzone
+    else:
+        local_offset_secs = -1 * time.timezone
+
+    base_time += local_offset_secs - tz_offset_secs
+    return base_time
+
+
+def parse_change_log():
```

Comment:
Do we have any (direct) (unit) tests for this function and for the one above for parsing the date string?

Reply:
There hasn't been any test before, but that's a good idea.